pycbc/waveform/bank.py

```diff
@@ -638,13 +639,60 @@ def __init__(self, filename, filter_length, delta_f, dtype,
         self.delta_t = 1.0 / (self.N * self.delta_f)
         self.filter_length = filter_length
         self.max_template_length = max_template_length
+        self.waveform_decompression_method = waveform_decompression_method
 
         super(FilterBank, self).__init__(filename, approximant=approximant,
             parameters=parameters, load_compressed=load_compressed,
             load_compressed_now=load_compressed_now,
             **kwds)
         self.ensure_standard_filter_columns(low_frequency_cutoff=low_frequency_cutoff)
 
+    def get_decompressed_waveform(self, tempout, index, f_lower=None,
+                                  approximant=None):
+        """Returns a frequency domain decompressed waveform for the template
+        in the bank corresponding to the index taken in as an argument. The
+        decompressed waveform is obtained by interpolating in frequency space,
+        the amplitude and phase points for the compressed template that are
+        read in from the bank."""
+
+        from pycbc.waveform.waveform import props
+        from pycbc.waveform import get_waveform_filter_length_in_time
+
+        # Get the precision used to generate the compressed template
+        precision_from_bank_file = self.compressed_waveforms[self.table.template_hash[index]].precision
+
+        # Get the precision that would be used to generate the decompressed
+        # waveform
+        waveform_decompression_precision = tempout.precision
+
+        if waveform_decompression_precision != precision_from_bank_file :
+            raise ValueError("The precision used to compress the "
+                             "waveform was %s, whereas the precision "
+                             "selected to decompress the waveform was "
+                             "%s. Both of theses should be the same. "
+                             "Therefore, use a bank with precision %s."
+                             %(precision_from_bank_file,
+                             waveform_decompression_precision,
+                             waveform_decompression_precision))
+
```

Comment:
@cdcapano Do you think it would be fine if I remove this condition check here? I had added this check when I was having problems decompressing waveforms when running pycbc_inspiral if the bank was made using precision double. But now that problem should be fixed by your fixes because the precision to do the decompression now would be decided by the precision of the out vector fed as input to `decompress` in `compress.py`. I just did a test to run pycbc-inspiral using a single template bank made with double precision and the waveform was decompressed properly. Should we still keep this check just to be safe?